File: Practice/13/C++/13. Простое число/13. Простое число/13. Простое число.cpp

#include <iostream>
#include <math.h>

int main()
{
    int n;
    bool flag = true;
    setlocale(LC_ALL, "ru");
    std::cout << "Введите число:" << std::endl; std::cin >> n;
    if (n >= 2) {
        for (int i = 2; i <= sqrt(n); i++) {
            if (n % i == 0) {
                flag = false;
                break;
            }
        }

        if (flag == true) {
            std::cout << "Простое" << std::endl;
        }
        else {
            std::cout << "Сложное" << std::endl;
        }
    }
    else if(n <= 1) {
        std::cout << "Число должно быть больше 1" << std::endl;
    }
}

File: Practice/13/Python/13. Простое число/13. Простое число/_13._Простое_число.py
import math
try:
    n = int(input("Введите число: "))
    i = 2
    limit = int(math.sqrt(n))
    flag=1
    if n < 2:
        print("Число должно быть больше 2")
    else:
        while i <= limit:
            if n % i == 0:
                print("Это составное число")
                flag=2
                
            i = i+1
        if flag==1:    
            print("Это простое число")
except ValueError:
    print("Ошибка ввода")
